### PR TITLE
fix(plate upload)

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: genotype_api/services/endpoint_services/plate_service.py [hunk 1]
```diff
@@ -73,26 +73,24 @@ def upload_plate(self, file: UploadFile) -> PlateResponse:
...
         analyses: list[Analysis] = list(excel_parser.generate_analyses())
         self.store.check_analyses_objects(analyses=analyses, analysis_type=Types.GENOTYPE)
         self.store.create_analyses_samples(analyses=analyses)
         plate_obj = Plate(plate_id=plate_id)
         plate_obj.analyses = analyses
         plate: Plate = self.store.create_plate(plate=plate_obj)
-        for analysis in plate.analyses:
-            self.store.refresh_sample_status(sample=analysis.sample)
+        for analysis in analyses:
+            sample: Sample = self.store.get_sample_by_id(sample_id=analysis.sample_id)
```
Comment (islean) [inline genotype_api/services/endpoint_services/plate_service.py:91]: what is the difference between this and analysis.sample?

**Review by islean — APPROVED** [review]
So this is more of a visual update? So that the user can see that the plate already exists?

File: genotype_api/api/endpoints/plates.py [hunk 2]
```diff
@@ -24,15 +24,20 @@ def get_plate_service(store: Store = Depends(get_store)) -> PlateService:
...
     file: UploadFile = File(...),
     plate_service: PlateService = Depends(get_plate_service),
     current_user: CurrentUser = Depends(get_active_user),
 ):
-    return plate_service.upload_plate(file)
+
+    try:
+        plate_service.upload_plate(file)
+    except PlateExistsError:
+        raise HTTPException(
```
Comment (seallard) [inline genotype_api/api/endpoints/plates.py:37]: I don't think we should be raising `HTTPExceptions` in the service layer. It does not belong here but should be done in the API layer where the response is created

File: genotype_api/api/endpoints/plates.py [hunk 3]
```diff
@@ -24,15 +24,20 @@ def get_plate_service(store: Store = Depends(get_store)) -> PlateService:
...
+            detail="Plate already exists in the database.", status_code=HTTPStatus.BAD_REQUEST
+        )
+    return JSONResponse("Plate uploaded successfully", status_code=status.HTTP_201_CREATED)
```
Comment (seallard) [inline genotype_api/api/endpoints/plates.py:40]: This belongs in the API layer too. In other places (like trailblazer), we probably should not call the data model returned from the service layer "...Response" and just use the model name, maybe with a `DTO` suffix.

It is better to keep all the response/request logic isolated to the API layer, something we have neglected in other places.

Feel free to skip changing it in this PR, just want to point it out so we are on the same page.

**Review by seallard — APPROVED** [review]
👍

File: genotype_api/api/endpoints/plates.py (hunk 2 above)
Comment (ChrOertlin) [reply genotype_api/api/endpoints/plates.py:37]: This is the api endpoint, I am not following.

File: genotype_api/api/endpoints/plates.py (hunk 3 above)
Comment (ChrOertlin) [reply genotype_api/api/endpoints/plates.py:40]: Ok then I am not sure I understand the whole structure, maybe we can sit down together at some point and figure this out.

File: genotype_api/services/endpoint_services/plate_service.py (hunk 1 above)
Comment (ChrOertlin) [reply genotype_api/services/endpoint_services/plate_service.py:95]: shouldn't be one. was just for testing I will revert

**Comment by ChrOertlin** [comment]
# Upload works:

image

File: genotype_api/api/endpoints/plates.py (hunk 3 above)
Comment (ChrOertlin) [reply genotype_api/api/endpoints/plates.py:40]: REsolving for now in favour of patching the bug

File: genotype_api/api/endpoints/plates.py (hunk 2 above)
Comment (ChrOertlin) [reply genotype_api/api/endpoints/plates.py:37]: Same here resolving in favour pof applying patch will open an issue

**Pull request merged**